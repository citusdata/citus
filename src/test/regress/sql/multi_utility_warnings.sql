--
-- MULTI_UTILITY_WARNINGS
--

-- Tests to check if we inform the user about potential caveats of creating new
-- databases, schemas, and roles.


SET citus.next_shard_id TO 1010000;


CREATE DATABASE new_database;

CREATE ROLE new_role;

CREATE USER new_user;

INSERT INTO pg_dist_authinfo VALUES (0, 'new_user', 'password=1234');

BEGIN;
INSERT INTO pg_dist_node VALUES (1234567890, 1234567890, 'localhost', 5432);
INSERT INTO pg_dist_poolinfo VALUES (1234567890, 'port=1234');
ROLLBACK;
