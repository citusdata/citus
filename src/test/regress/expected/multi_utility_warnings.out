--
-- MULTI_UTILITY_WARNINGS
--
-- Tests to check if we inform the user about potential caveats of creating new
-- databases, schemas, and roles.
SET citus.next_shard_id TO 1010000;
CREATE DATABASE new_database;
NOTICE:  Citus partially supports CREATE DATABASE for distributed databases
DETAIL:  Citus does not propagate CREATE DATABASE command to workers
HINT:  You can manually create a database and its extensions on workers.
CREATE ROLE new_role;
NOTICE:  not propagating CREATE ROLE/USER commands to worker nodes
HINT:  Connect to worker nodes directly to manually create all necessary users and roles.
CREATE USER new_user;
NOTICE:  not propagating CREATE ROLE/USER commands to worker nodes
HINT:  Connect to worker nodes directly to manually create all necessary users and roles.
INSERT INTO pg_dist_authinfo VALUES (0, 'new_user', 'password=1234');
ERROR:  cannot write to pg_dist_authinfo
DETAIL:  Citus Community Edition does not support the use of custom authentication options.
HINT:  To learn more about using advanced authentication schemes with Citus, please contact us at https://citusdata.com/about/contact_us
BEGIN;
INSERT INTO pg_dist_node VALUES (1234567890, 1234567890, 'localhost', 5432);
INSERT INTO pg_dist_poolinfo VALUES (1234567890, 'port=1234');
ERROR:  cannot write to pg_dist_poolinfo
DETAIL:  Citus Community Edition does not support the use of pooler options.
HINT:  To learn more about using advanced pooling schemes with Citus, please contact us at https://citusdata.com/about/contact_us
ROLLBACK;
